docs/_docs/key-value-api/read-repair.adoc

```diff
@@ -0,0 +1,99 @@
+
+To enable the `Read Repair` mode, obtain an instance of the cache that enables Read Repair reads as follows:
+
+[source, java]
+----
+include::{javaCodeDir}/BasicCacheOperations.java[tags=read-repair, indent=0]
+----
+
+== Strategies
+In case consistency violations were found, the values across the topology will be replaced by repaired values according to the chosen strategy:
+
+[cols="1,4",opts="header"]
+|===
+| Strategy | Description
+| `LWW` a| Last write (the newest entry) wins.
+
+May cause IgniteException when the fix is impossible (unable to detect the newest entry)::
+* Null(s) found as well as non-null values for the same key. Null (missed entry) has no version, so, it can not be compared with the versioned entry.
+* Entries with the same version have different values.
+| `PRIMARY` | Value from the primary node wins.
+| `RELATIVE_MAJORITY` a| The relative majority, any value found more times than any other wins.
+Works for an even number of copies (which is typical of Ignite) instead of an absolute majority.
+
+May cause IgniteException when unable to detect values found more times than any other.
+
+For example, when we have 5 copies (4 backups)::
+* and value `A` found twice, but `X`, `Y` and `Z` only once, `A` wins,
+* but, when `A` is found twice, as well as `B`, and `X` only once, the strategy is unable to determine the winner.
+
+When we have 4 copies (3 backups), any value found two or more times, when others are found only once, is the winner.
+| `REMOVE` | Inconsistent entries will be removed.
+| `CHECK_ONLY` | Only check will be performed.
+|===
+
+== Events
+link:https://ignite.apache.org/releases/{version}/javadoc/org/apache/ignite/events/EventType.html#EVT_CONSISTENCY_VIOLATION[Сonsistency Violation Event] will be recorded for each violation in case it's configured as recordable. You can listen to this event to get notified about inconsistency issues.
+
+Refer to the link:events/listening-to-events[Working with Events] section for the information on how to listen to events.
+
+== Transactional Caches
+Values will be repaired::
+* automatically for transactions that have `TransactionConcurrency.OPTIMISTIC` concurrency mode or `TransactionIsolation.READ_COMMITTED` isolation level,
+
+* at commit() phase for transactions that have `TransactionConcurrency.PESSIMISTIC` concurrency mode and isolation level other than `TransactionIsolation.READ_COMMITTED`
+
+[WARNING]
+====
+[discrete]
+=== Limitations
+This proxy usage does not guarantee "all copies check" in case the value is already cached inside the transaction.
+
+In case you do use not a `READ_COMMITTED` isolation mode and already have a cached value, for example already read the value or performed a write, you'll just gain the cached value.
```

Comment:
Too many `already`
`If you don't use a READ_COMMITTED`
`have already read the value or performed a write`
`you will just get` (not sure what you meant here)

Reply:
it means you'll gain a cached value